feat(grenades): Add grenade events to json

LarsKoelpin opened the pull request:
Hey there,
as stated in #32 it would be nice to have grenade events in the resulting JSON.
So I made the fixes for myself.

- [x]  Add grenade events to JSON
- [x] Add a testcase
- [x] Review changes
- [x] Update docs
As I haved worked with Go before, It would be nice to have my changes reviewed

markus-wa commented:
Nice one @LarsKoelpin - thanks for the contribution. this will need some more work for protobuf it seems, but looks pretty good otherwise.

see:
- https://github.com/markus-wa/cs-demo-minifier/blob/master/protobuf/gen/proto/replay.proto#L65
- https://github.com/markus-wa/cs-demo-minifier#generating-protobuf-code
- https://github.com/markus-wa/cs-demo-minifier/blob/master/protobuf/protobuf.go#L39

You'll also have to update the `.golden` files:
https://github.com/markus-wa/cs-demo-minifier/blob/master/README.md#updating-golden-files

markus-wa commented on events.go:
```diff
@@ -1,9 +1,9 @@
-	rep "github.com/markus-wa/cs-demo-minifier/replay"
-	dem "github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs"
-	events "github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs/events"
+  rep "github.com/markus-wa/cs-demo-minifier/replay"
+  dem "github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs"
+  events "github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs/events"
```
```suggestion
	rep "github.com/markus-wa/cs-demo-minifier/replay" 
	dem "github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs"
	events "github.com/markus-wa/demoinfocs-golang/v2/pkg/demoinfocs/events"
```

formatting issue

markus-wa commented on events.go:
```diff
@@ -175,6 +176,73 @@ func (defaultEventHandlers) RegisterChatMessage(ec *EventCollector) {
+    eb.floatAttr("x", e.Position.X)
+    eb.floatAttr("y", e.Position.Y)
+    eb.floatAttr("z", e.Position.Z)
+    eb.intAttr(rep.AttrKindEntityID, e.Thrower.EntityID);
```
I think we will want a new `AttrKind`

probably `AttrKindThrowerID`

because there are technically also grenade entities. When I used the word `entity` for players in other parts was probably a mistake by me, but that's a topic for another day.

LarsKoelpin replied:
So, how should the resulting JSON look like? We currently have something like this.

```
{"name":"smoke_start","attrs":[{"key":"x","numVal":-291.2807312011719},{"key":"y","numVal":1448.5047607421875},{"key":"z","numVal":-26.500001907348633},{"key":"entityId","numVal":14}]}
```

I guess the goal  should be something like this:
```json
{"name":"smoke_start","attrs":[{"key":"x","numVal":-291.2807312011719},{"key":"y","numVal":1448.5047607421875},{"key":"z","numVal":-26.500001907348633},{"key":"throwerEntityId","numVal":14}]}
```
(Renamed EntityID to throwerId)
Or more like this?
```json
{
"name":"smoke_start",
"attrs":[
  { 
   "key":"position",
   "objVal": {"x": -291.2807312011719, "y": 1448.5047607421875, "z":26.500001907348633 }]}
```
(with an object value type)

markus-wa replied:
I might have gone with `throwerId` for the sake of shortness but `throwerEntityId` is good as well 👍

LarsKoelpin commented:
* I renamend the events to past tense "started" instead of "start" to conform with the other events already present.
* I renamed the entityId to throwerEntityId
* I added documentation to the events
* The shape of the events looks like this:
```json
	{
					"name": "smoke_expired",
					"attrs": [
						{
							"key": "x",
							"numVal": 920.5975341796875
						},
						{
							"key": "y",
							"numVal": 269.741943359375
						},
						{
							"key": "z",
							"numVal": 1616.0150146484375
						},
						{
							"key": "throwerEntityId",
							"numVal": 9
						}
					]
				}
```

I'm not sure, if I migrated the protobuf related files correctly. I'd be fine with my changes otherwise

markus-wa commented:
wooops, I just noticed I had misconfigured the linter.

reviewdog has some findings now. after that I'll give it another look but it looks good :slightly_smiling_face:

markus-wa commented on events.go:
```diff
@@ -1,7 +1,7 @@
 
 import (
-	rep "github.com/markus-wa/cs-demo-minifier/replay"
+	rep "github.com/markus-wa/cs-demo-minifier/replay" 
```
![image](https://user-images.githubusercontent.com/5138316/83359301-54596c80-a371-11ea-9e36-96f64721b176.png)

LarsKoelpin replied:
:D

markus-wa replied:
it can! :slightly_smiling_face: with the help of the 'Save Actions' plugin:

![image](https://user-images.githubusercontent.com/5138316/83360323-4529ed00-a378-11ea-8357-0badada79dd4.png)

LarsKoelpin replied:
thanks :)

markus-wa commented:
oh, and I think you might have forgotten to commit some changes to `protobuf/gen/proto/replay.proto`

LarsKoelpin commented:
finally

markus-wa commented:
nice one, great work!

markus-wa commented:
merged and released, thanks for the contribution :partying_face: 

https://github.com/markus-wa/cs-demo-minifier/releases/tag/v0.4.1
